### PR TITLE
Add auto build + push of docker images on tag

### PR DESCRIPTION
**Pull request by Kay-Zee**
Will auto build the docker images on a new git tag

### CODE REVIEW

File: integration/testnet/container.go [hunk 1]
```diff
@@ -33,9 +43,9 @@ type ContainerConfig struct {
 // ImageName returns the Docker image name for the given config.
 func (c *ContainerConfig) ImageName() string {
 	if c.Ghost {
-		return "gcr.io/dl-flow/ghost:latest"
+		return defaultRegistry + "/ghost:latest"
 	}
-	return fmt.Sprintf("gcr.io/dl-flow/%s:latest", c.Role.String())
+	return fmt.Sprintf(defaultRegistry+"/%s:latest", c.Role.String())
```
Comment (psiemens) [inline integration/testnet/container.go:48]:
```suggestion
	return fmt.Sprintf("%s/%s:latest", defaultRegistry, c.Role.String())
```

**Review by psiemens — APPROVED** [review]

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    name: Push to container registry
+    runs-on: ubuntu-latest
+    steps:
+    - name: Granting private modules access
+      run: |
+        git config --global url."https://${{ secrets.GO_MODULES_TOKEN }}:x-oauth-basic@github.com/onflow".insteadOf "https://github.com/onflow"
+    - name: Setup Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: '1.14'
```
Comment (zhangchiqing) [inline .github/workflows/cd.yml:19]: Do we want to be consistent with the version in [go.mod](https://github.com/onflow/flow-go/blob/master/go.mod#L3)?

File: .github/workflows/cd.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+name: CD
+
+on:
+  push:
```
Comment (zhangchiqing) [inline .github/workflows/cd.yml:4]: What if we force push a tag to overwrite, would an image be rebuilt and updated?

**Review by zhangchiqing — APPROVED** [review]
Nice 👍

File: .github/workflows/cd.yml (hunk 3 above)
Comment (Kay-Zee) [reply .github/workflows/cd.yml:4]: I believe so, yes

**Pull request merged**